portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,12 +41,26 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
+    String strNumComments = request.getParameter("numComments");
+    int numComments;
+    try{ 
+      numComments = Integer.parseInt(strNumComments);
+      if(numComments < 0) {
+          throw new Exception();
```

Comment:
it's good practice to write some kind of message into the exception (`new Exception("Invalid number of comments.")` perhaps)